apps/emqx/src/emqx_secret.erl

```diff
@@ -29,13 +29,38 @@
 %% API funcions
 %%================================================================================
 
+%% @doc Wrap a term in a secret closure.
+%% This effectively hides the term from any term formatting / printing code.
+-spec wrap(T) -> t(T).
 wrap(Term) ->
     fun() ->
         Term
     end.
 
+%% @doc Wrap a function call over a term in a secret closure.
+%% This is slightly more flexible form of `wrap/1` with the same basic purpose.
+-spec wrap(module(), atom(), _Term) -> t(_).
+wrap(Module, Function, Term) ->
+    fun() ->
+        apply(Module, Function, [Term])
+    end.
+
+%% @doc Unwrap a secret closure, revealing the secret.
+%% This is either `Term` or `Module:Function(Term)` depending on how it was wrapped.
+-spec unwrap(t(T)) -> T.
 unwrap(Term) when is_function(Term, 0) ->
     %% Handle potentially nested funs
     unwrap(Term());
 unwrap(Term) ->
     Term.
+
+%% @doc Inspect the term wrapped in a secret closure.
+-spec term(t(_)) -> _Term.
+term(Wrap) when is_function(Wrap, 0) ->
+    case erlang:fun_info(Wrap, module) of
+        {module, ?MODULE} ->
+            {env, Env} = erlang:fun_info(Wrap, env),
```

Comment:
this is a great idea.
we can keep the conventional arity-0 wrapper but also make possible to inspect other values.

If we are to make `file://` conventional for EMQX secret files, maybe no need to introduce another module, just implement everything here in this module.

for example:
```
wrap(<<"file://", Path/binary>>) ->
    Source = ?PASSWORD_FILE(Path) ,
    fun() ->
            Path1 = element(2, Source),
            read_secret_file(Path1)
    end;
wrap(Secret) ->
    fun() -> Secret end.

unwrap(Term) when is_function(Term, 0) ->
    %% Handle potentially nested funs
    unwrap(Term());
unwrap(Term) ->
    Term.

source(Wrap) ->
    case erlang:fun_info(Wrap, env) of
        {env, [F]} when is_function(F) ->
            source(F);
        {env, [?PASSWORD_FILE(Path)]} ->
            <<"file://", Path/binary>>;
        {env, [Secret]} ->
            Secret
    end.
```

Reply:
> maybe no need to introduce another module, just implement everything here in this module.

The reason for a very generic `wrap/3` and a separate module is that I wanted to ensure that `emqx_secret` module won't see any changes for sufficiently long time, due to hot-upgrade concerns.